### PR TITLE
entgql: Unmarshal whereInput using graphql package

### PR DESCRIPTION
**Pull request by giautm**
Related to https://github.com/ent/ent/issues/2351

### CODE REVIEW

File: entgql/template/collection.tmpl [hunk 1]
```diff
@@ -166,8 +166,13 @@ func {{ $newPaginateArg }}(rv map[string]interface{}) *{{ $paginateArg }} {
 		{{- $withWhere := true }}{{ with $node.Annotations.EntGQL }}{{ if isSkipMode .Skip "where_input" }}{{ $withWhere = false }}{{ end }}{{ end }}
 		{{- if $withWhere }}
 			{{- $where := $names.WhereInput }}
-			if v := rv[whereField]; v != nil && v != (*{{ $where }})(nil) {
-				args.opts = append(args.opts, {{ $filter }}(v.(*{{ $where }}).Filter))
+			if v, ok := rv[whereField]; ok {
+				switch v := v.(type) {
```
Comment (a8m) [inline entgql/template/collection.tmpl:170]: Why not if?

Comment (giautm) [reply entgql/template/collection.tmpl:170]: Just make sure it doesn't panic in any case. 🙃

Comment (giautm) [reply entgql/template/collection.tmpl:170]: Ah, Got it.

**Review by a8m — APPROVED** [review]

**Comment by giautm** [comment]
Release to the 🌔 ,

**Pull request merged**